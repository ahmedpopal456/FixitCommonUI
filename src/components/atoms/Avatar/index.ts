import { AvatarBadge } from './Avatar';
export { AvatarBadge};