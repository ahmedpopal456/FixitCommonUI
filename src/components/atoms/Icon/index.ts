import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
