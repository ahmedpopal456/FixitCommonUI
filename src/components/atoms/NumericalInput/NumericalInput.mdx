---
name: NumericalInput
menu: Atoms
---
import { Props, Playground } from "docz";
import { NumericalInput } from "./NumericalInput.tsx";
import { DoczContainer } from '../../../utils/DoczContainer.tsx'

# Numerical Input
A Numerical Input is a clickable view to enter numerical values into a field.

# Usage
```tsx
import { NumericalInput } from "fixit-common-ui";
<NumericalInput numbers={1}/>
```

## Props
<Props of={NumericalInput}></Props>

## Examples
<Playground>
  <DoczContainer>
    <NumericalInput numbers={4}/>
    <NumericalInput numbers={3} borderColor="accent"/>
  </DoczContainer>
</Playground>

## Note
You can enter multiple digits in the web browser. 
But on mobile, there is a max length of 1 as input and it opens the numpad instead of an alphabetical keyboard.
