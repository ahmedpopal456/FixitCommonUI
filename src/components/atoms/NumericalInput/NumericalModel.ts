import colors from '../../../theme/colors';

export interface NumericalProps {
    /** Can be any color from the theme color palette */
    color?: keyof typeof colors;
    /** Can be any color from the theme color palette */
    borderColor?: keyof typeof colors;
    /** Specify the number of Numerical Inputs to be created in a row */
    numbers: number;
  }
