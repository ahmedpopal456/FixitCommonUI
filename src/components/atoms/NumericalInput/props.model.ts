import colors from "../../../theme/colors";
  
  export interface Props {
    /** Can be any color from the theme color palette */
    color?: keyof typeof colors;
    /** Can be any color from the theme color palette */
    borderColor?: keyof typeof colors;
  }