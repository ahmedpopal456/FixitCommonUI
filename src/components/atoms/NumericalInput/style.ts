import styled from "styled-components/native";
import { Props } from './props.model';
import colors from "../../../theme/colors";


/** 
 * Styled Components
 * 
 * Here you can add css to react native components to create a styled version of it
 */
export const NumericalInputContainer = styled.View`
    align-self: center;
    justify-content: space-around;
    flex-direction: row;
    margin: 5px;
    padding: 7px;
    height: 50px;
    background-color: white;
    ${(props: Props) =>
        `
        border-color: ${colors[props.borderColor!]};
        border-width: 1px;
    `}
`;
