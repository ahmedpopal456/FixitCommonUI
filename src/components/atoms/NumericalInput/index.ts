import { NumericalInput } from './NumericalInput';

export { NumericalInput };
