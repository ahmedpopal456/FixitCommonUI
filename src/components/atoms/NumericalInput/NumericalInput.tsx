import React, { useState } from "react";
import { NumericalInputContainer, StyledNumInput } from "./style"
import { NumericalProps } from "./NumericalProps.model";

/** 
 * Describe the Component 
 * @param props - The NumericalInput's props
 * @returns A react native custom NumericalInput component
 */

export const NumericalInput: React.FC<NumericalProps> = (props: NumericalProps): JSX.Element => {
  var arr = new Array(props.numbers);
  for (let i = 0; i < arr.length; i++)
    arr[i] = "NumInputKey_" + i;

  return (
    <div style={{display: 'flex', flexDirection: 'row'}}>
      {arr.map(function(index) {
        return <NumericalInputContainer
          key={index}
          borderColor={props.borderColor}
          numbers={props.numbers}
        >
          <StyledNumInput
            {...props}
            keyboardType="number-pad"
            maxLength={1} // only works on mobile devices!!
            style={{fontSize: 25}}
          />
        </NumericalInputContainer>;
      })}
    </div>
    
    // <NumericalInputContainer
    //   borderColor={props.borderColor}
    //   numbers={props.numbers}
    // >
    //   <StyledNumInput
    //     {...props}
    //     keyboardType="number-pad"
    //     maxLength={1} // only works on mobile devices!!
    //     style={{fontSize: 25}}
    //   />
    // </NumericalInputContainer>
  );
};

/** Setting the defaults for the props*/
NumericalInput.defaultProps = {
  numbers: 4,
  borderColor: "primary"
};
    