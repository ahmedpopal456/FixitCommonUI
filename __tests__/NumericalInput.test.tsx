import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { NumericalInput } from '../src/components/atoms/NumericalInput';

it('matches default snapshot', () => {
  const numinput = renderer.create(<NumericalInput numbers={1}/>);
  expect(numinput.toJSON()).toMatchSnapshot();
});

it('matches border color snapshot', () => {
  const numinput = renderer.create(<NumericalInput numbers={1} borderColor="accent"/>);
  expect(numinput.toJSON()).toMatchSnapshot();
});

it('matches the number of inputs snapshot', () => {
  const numinput = renderer.create(<NumericalInput numbers={4}/>);
  expect(numinput.toJSON()).toMatchSnapshot();
});
