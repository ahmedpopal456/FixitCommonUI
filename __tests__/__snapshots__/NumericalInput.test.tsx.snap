// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches border color snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    borderColor="accent"
    numbers={1}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#FFD14A",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="accent"
      keyboardType="number-pad"
      maxLength={1}
      numbers={1}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#FFD14A",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</div>
`;

exports[`matches default snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    borderColor="primary"
    numbers={1}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#33343E",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="primary"
      keyboardType="number-pad"
      maxLength={1}
      numbers={1}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#33343E",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</div>
`;

exports[`matches the number of inputs snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    borderColor="primary"
    numbers={4}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#33343E",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="primary"
      keyboardType="number-pad"
      maxLength={1}
      numbers={4}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#33343E",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    borderColor="primary"
    numbers={4}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#33343E",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="primary"
      keyboardType="number-pad"
      maxLength={1}
      numbers={4}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#33343E",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    borderColor="primary"
    numbers={4}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#33343E",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="primary"
      keyboardType="number-pad"
      maxLength={1}
      numbers={4}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#33343E",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    borderColor="primary"
    numbers={4}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "#33343E",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingBottom": 7,
          "paddingLeft": 7,
          "paddingRight": 7,
          "paddingTop": 7,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="primary"
      keyboardType="number-pad"
      maxLength={1}
      numbers={4}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#33343E",
            "textAlign": "center",
            "width": 30,
          },
          Object {
            "fontSize": 25,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</div>
`;
